packages/pipeline-editor/src/SubmitNotebook.ts

```diff
@@ -73,52 +73,33 @@ export class SubmitNotebookButtonExtension
 
   readonly app: JupyterFrontEnd;
 
-  showWidget = (): void => {
+  showWidget = async (): Promise<void> => {
     const envVars: string[] = NotebookParser.getEnvVars(
       this.panel.content.model.toString()
     );
 
-    SubmissionHandler.makeGetRequest(
-      'api/metadata/runtimes',
-      'pipeline',
-      (response: any) => {
-        if (Object.keys(response.runtimes).length === 0) {
-          return SubmissionHandler.noMetadataError('runtimes');
-        }
-
-        FrontendServices.getRuntimeImages().then(
-          (runtimeImages: IDictionary<string>) => {
-            showDialog({
-              title: 'Submit notebook',
-              body: new SubmitNotebook(
-                envVars,
-                response.runtimes,
-                runtimeImages
-              ),
-              buttons: [Dialog.cancelButton(), Dialog.okButton()]
-            }).then(result => {
-              if (result.value == null) {
-                // When Cancel is clicked on the dialog, just return
-                return;
-              }
-
-              // prepare notebook submission details
-              const notebookOptions: ISubmitNotebookOptions = result.value as ISubmitNotebookOptions;
-              const pipeline = Utils.generateNotebookPipeline(
-                this.panel.context.path,
-                notebookOptions
-              );
-
-              SubmissionHandler.submitPipeline(
-                pipeline,
-                result.value.runtime_config,
-                'notebook'
-              );
-            });
-          }
-        );
+    const runtimes = await FrontendServices.getMetadata('runtimes');
```

Comment:
`getMetadata('runtimes')` is called from `handleExport()` and `handleRun()`, as well as here.  I can't tell if `FrontendServices.getMetadata()` unconditionally issues a request to the server or not, but its good that we have a central place in case we want to cache metadata in general.

Reply:
The original intention of #271 was to create a service class for runtimes, so I initially started writing a function like `getRuntimeImages` but I stop when I realized that it would just be:
```
static async getRuntimes(): Promise<any> {
    const runtimes = await FrontendServices.getMetadata('runtimes');
    return runtimes;
}
```
and I felt that was silly. Unlike our other metadata runtimes returns already in the format we need to use it.

Since we currently don't cache metadata I didn't think it was worth it, but if we want to enable an easy way to do that in the future I can create the `getRuntimes()` function in SubmissionHandler to be used in those the there uses.

Reply:
Why wouldn't we just add the caching into `getMetadata` and allow all metadata namespaces to use the same caching framework - a cache per namespace - when that time comes?